File: app/XDM/MediaParser.Hls/Dash/Representation.cs
using System;
using System.Collections.Generic;

namespace MediaParser.Dash
{
    public class Representation
    {
        public int Width { get; }
        public int Height { get; }
        public string Codec { get; }
        public long Bandwidth { get; }
        public long Duration { get; }
        public List<Uri> Segments { get; }
        public string MimeType { get; }
        public string Language { get; }

        public Representation(List<Uri> segments, int width, int height,
            string codec, long bandwidth, long duration, string mimeType, string language)
        {
            this.Segments = segments;
            this.Width = width;
            this.Height = height;
            this.Codec = codec;
            this.Bandwidth = bandwidth;
            this.Duration = duration;
            this.MimeType = mimeType;
            this.Language = language;
        }
    }
}

File: xdm-core/src/main/java/xdm/core/media/parser/dash/Representation.java
package xdm.core.media.parser.dash;

import java.net.URI;
import java.util.List;
import lombok.Getter;
import lombok.ToString;

@ToString
@Getter
public class Representation {
  private final int width;
  private final int height;
  private final String codec;
  private final long bandwidth;
  private final long duration;
  @ToString.Exclude private final List<URI> segments;
  private final String mimeType;
  private final String language;

  public Representation(
      List<URI> segments,
      int width,
      int height,
      String codec,
      long bandwidth,
      long duration,
      String mimeType,
      String language) {
    this.segments = segments;
    this.width = width;
    this.height = height;
    this.codec = codec;
    this.bandwidth = bandwidth;
    this.duration = duration;
    this.mimeType = mimeType;
    this.language = language;
  }
}
